### PR TITLE
Gh5802 Pricelist mutation and product proposal changes

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/5802

### CODE REVIEW

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.business/src/main/java/de/metas/pricing/callout/M_PricelistVersion_TabCallout.java  2
- de.metas.business/src/test/java/de/metas/pricing/service/impl/PriceListDAOTest.java  2
- de.metas.business/src/main/java/de/metas/pricing/interceptor/M_PriceList.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4569860&bid=15361535)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.business/src/test/java/de/metas/pricing/service/impl/PriceListDAOTest.java  2
- de.metas.business/src/main/java/de/metas/pricing/interceptor/M_PriceList.java  6
- de.metas.business/src/main/java/de/metas/pricing/callout/M_PricelistVersion_TabCallout.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4569860&bid=15361535)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.business/src/test/java/de/metas/pricing/service/impl/PriceListDAOTest.java  2
- de.metas.business/src/main/java/de/metas/pricing/interceptor/M_PriceList.java  6
- de.metas.business/src/main/java/de/metas/pricing/callout/M_PricelistVersion_TabCallout.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4569860&bid=15361535)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.business/src/main/java/de/metas/pricing/callout/M_PricelistVersion_TabCallout.java  2
- de.metas.business/src/test/java/de/metas/pricing/service/impl/PriceListDAOTest.java  2
- de.metas.business/src/main/java/de/metas/pricing/interceptor/M_PriceList.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4569860&bid=15361535)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.business/src/main/java/de/metas/pricing/callout/M_PricelistVersion_TabCallout.java  2
- de.metas.business/src/main/java/de/metas/pricing/interceptor/M_PriceList.java  6
- de.metas.business/src/test/java/de/metas/pricing/service/impl/PriceListDAOTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4569860&bid=15361535)

File: de.metas.business/src/main/java/de/metas/pricing/service/impl/PriceListDAO.java [hunk 1]
```diff
@@ -695,11 +710,29 @@ private void createNewPLV(final I_M_PriceList_Version oldCustomerPLV, final I_M_
 
 		cloneASIs(newCustomerPLVId);
 
-		newCustomerPLV.setM_Pricelist_Version_Base_ID(newBasePLV.getM_Pricelist_Version_Base_ID());
 		save(newCustomerPLV);
 
 	}
 
+	@Override
+	public String createPLVName(final PriceListId pricelistId, LocalDate date)
```
Comment (teosarca) [inline de.metas.business/src/main/java/de/metas/pricing/service/impl/PriceListDAO.java]: pls null check in case `date` null

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.business/src/test/java/de/metas/pricing/service/impl/PriceListDAOTest.java  2
- de.metas.business/src/main/java/de/metas/pricing/callout/M_PricelistVersion_TabCallout.java  4
- de.metas.business/src/main/java/de/metas/pricing/interceptor/M_PriceList.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4569860&bid=15361535)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.business/src/test/java/de/metas/pricing/service/impl/PriceListDAOTest.java  2
- de.metas.business/src/main/java/de/metas/pricing/callout/M_PricelistVersion_TabCallout.java  4
- de.metas.business/src/main/java/de/metas/pricing/interceptor/M_PriceList.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4569860&bid=15361535)

**Pull request merged**